server/datastore/mysql/software_installers.go

```diff
@@ -287,3 +323,98 @@ WHERE
 
 	return &dest, nil
 }
+
+func tmplNamedSQLCaseHostSoftwareInstallStatus(alias string) string {
+	return fmt.Sprintf(`
+	CASE WHEN %[1]s.post_install_script_exit_code IS NOT NULL
+		AND %[1]s.post_install_script_exit_code = 0 THEN
+		:installed
+	WHEN %[1]s.post_install_script_exit_code IS NOT NULL
+		AND %[1]s.post_install_script_exit_code != 0 THEN
+		:failed
+	WHEN %[1]s.install_script_exit_code IS NOT NULL
+		AND %[1]s.install_script_exit_code = 0 THEN
+		:installed
+	WHEN %[1]s.install_script_exit_code IS NOT NULL
+		AND %[1]s.install_script_exit_code != 0 THEN
+		:failed
+	WHEN %[1]s.pre_install_query_output IS NOT NULL
+		AND %[1]s.pre_install_query_output = '' THEN
+		:failed
+	WHEN %[1]s.host_id IS NOT NULL THEN
+		:pending
+	ELSE
+		NULL -- not installed from Fleet installer
+	END`, alias)
+}
+
+func (ds *Datastore) GetSummaryHostSoftwareInstalls(ctx context.Context, installerID uint) (*fleet.SoftwareInstallerStatusSummary, error) {
+	var dest fleet.SoftwareInstallerStatusSummary
+
+	stmt := fmt.Sprintf(`
+SELECT
+	COALESCE(SUM( IF(status = :pending, 1, 0)), 0) AS pending,
+	COALESCE(SUM( IF(status = :failed, 1, 0)), 0) AS failed,
+	COALESCE(SUM( IF(status = :installed, 1, 0)), 0) AS installed
+FROM (
+SELECT
+	software_installer_id,
+	%s AS status
+FROM
+	host_software_installs hsi
+WHERE
+	software_installer_id = :installer_id
+	AND id IN(
+		SELECT
+			max(id) -- ensure we use only the most recently created install attempt for each host
+			FROM host_software_installs
+		WHERE 
+			software_installer_id = :installer_id
+		GROUP BY
+			host_id)) s`, tmplNamedSQLCaseHostSoftwareInstallStatus("hsi"))
+
+	query, args, err := sqlx.Named(stmt, map[string]interface{}{
+		"installer_id": installerID,
+		"pending":      fleet.SoftwareInstallerPending,
+		"failed":       fleet.SoftwareInstallerFailed,
+		"installed":    fleet.SoftwareInstallerInstalled,
+	})
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get summary host software installs: named query")
+	}
+
+	err = sqlx.GetContext(ctx, ds.reader(ctx), &dest, query, args...)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get summary host software install status")
```

Comment:
It should probably handle the `ErrNoRows` case to return a `notFound` error. Or would it ever happen since it's a SUM on a sub-select...